source/javascripts/cookie-consent.js

```diff
@@ -0,0 +1,108 @@
+window.GOVUK = window.GOVUK || {}
+
+window.GOVUK.gtm = {
+  init: function () {
+    this.consent = false;
+    window.dataLayer = window.dataLayer || [];
+    this.gtag();
+    this.checkConsent();
+    this.cookieForm = document.querySelector('[data-module="cookie-settings"]');
+    if (this.cookieForm) {
+      this.setupCookieForm();
+    }
+  },
+
+  gtag: function () {
+    dataLayer.push(arguments);
+  },
+
+  setupGtm: function () {
+    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+    })(window,document,'script','dataLayer','GTM-TNKCK97');      
+  },
+
+  checkConsent: function () {
+    var cookies = document.cookie;
+    cookies = cookies.split(';');
+
+    for (var i = 0; i < cookies.length; i++) {
+      var thisCookie = cookies[i].split('=');
+      if (thisCookie[0] == 'analytics_consent') {
+        // consent cookie exists and is yes, init gtm
+        // if it exists and is no, we just hide the banner
+        if (thisCookie[1] == 'true') {
+          this.consent = true;
+          this.setupGtm();
+        }
+        this.hideCookieBanner();
+        break;
+      }
+    }
+  },
+
+  denyCookies: function () {
+    this.consent = false;
+    this.gtag('consent', 'default', {
+      'ad_storage': 'denied'
+    });
+    this.cookieConsent();
+    this.hideCookieBanner();
```

Comment:
```suggestion
    this.hideCookieBanner();
    
    window.google_tag_manager = null
    window.google_tag_data = null
    document.cookie = '_ga=; Max-Age=0; path=/; domain=' + location.hostname;
    document.cookie = '_ga_EYTJMCJ2LP=; Max-Age=0; path=/; domain=' + location.hostname;
    var gtmScript1 = document.querySelector('script[src="https://www.googletagmanager.com/gtm.js?id=GTM-TNKCK97"]')
    var gtmScript2 = document.querySelector('script[src="https://www.googletagmanager.com/gtag/js?id=G-EYTJMCJ2LP&l=dataLayer&cx=c"]')
    if(gtmScript1) {
      gtmScript1.remove()
    }
    if(gtmScript2) {
      gtmScript2.remove()
    }
```

This should remove the cookies successfully. It clears the Google script from the window, sets the cookies to remove, and deletes the script from the head of the page. Up to you if you want to implement it, but thought it's best to make sure the rejection works properly.

Reply:
That's very thorough. Maybe a bit too thorough as if you then navigate to another page those scripts will be removed anyway? Also I wasn't certain whether the script would always have that exact name, although maybe it's based on the property ID.

Reply:
@andysellick Yeah I guess it was a temporary solution, but it looks like you can remove the last part and it will still work. I think the ID would stay the same (I worked on National Archives' cookie mechanism, and the ID never changed.) This may be cleane:r:

```JavaScript
    window.google_tag_manager = null
    window.google_tag_data = null
    document.cookie = '_ga=; Max-Age=0; path=/; domain=' + location.hostname;
    document.cookie = '_ga_EYTJMCJ2LP=; Max-Age=0; path=/; domain=' + location.hostname;
```

The issue seems to be that if you just delete the cookie, GTM is still running on the actual cookie page, so it sets a cookie again after deleting the cookie. But `window.google_tag_manager = null  window.google_tag_data = null` seems to solve that.